Yatima/Typechecker/Equal.lean

```diff
@@ -69,21 +76,24 @@ mutual
         equalThunks lvl args args'
       else pure false
     | .app (.proj idx val) args, .app (.proj idx' val') args' =>
```

Comment:
At first glance it's not clear to me how this case can arise. Shouldn't (type-correct) projections always be able to be reduced? Some docs would be helpful here.